modules/pricer/src/main/java/com/opengamma/strata/pricer/fx/BlackVolatilitySmileFxProvider.java

```diff
@@ -120,12 +120,12 @@ public SurfaceCurrencyParameterSensitivity surfaceParameterSensitivity(FxOptionS
     double forward = currencyPair.isInverse(point.getCurrencyPair()) ? 1d / point.getForward() : point.getForward();
     double pointValue = point.getSensitivity();
     DoubleMatrix bucketedSensi = smile.getVolatilityAndSensitivities(expiryTime, strike, forward).getSensitivities();
-    double[] times = smile.getTimeToExpiry();
+    double[] times = smile.getTimeToExpiry().toArray();
```

Comment:
Aim is to avoid doing this. Local variable should be of type `DoubleArray` and usages should be fixed.